src/modules/esl-image/core/esl-image.ts

```diff
@@ -89,7 +89,14 @@ export class ESLImage extends ESLBaseElement {
   }
 
   protected attributeChangedCallback(attrName: string, oldVal: string, newVal: string): void {
-    if (!this.connected || oldVal === newVal) return;
+    if (oldVal === newVal) return;
+    if (attrName === 'mode') {
+      this.syncImage(oldVal);
```

Comment:
It sounds like not a good idea to initialize the old strategy. We should use browser resources for actions that make the actual state of the component.

I see that there may be context loss in the case of cloning node, but it sounds like custom elements should give us logical and definite steps to make a new instance alive (re-init it). Or we can choose some more stable means like lazy getters.

The main problem right now (to provide more visibility to what I'm trying to say) is that we have extra operations in the normal cases, and we have it in a place that is not related to the issue directly.

Reply:
The problem breaks into two parts issue with cloning element and issue with re-appending.
Cloning element:
-Strategy context lost, meaning previous rules sometimes can't be cleared and new ones added to it. Fix for it is pushed is currently pushed in the branch
Re-appending element:
- Property `this._srcRules` has image source srtored in it. And because we are not capturing the attribute change before connected state, `this._srcRules` doesn't update, so it gets stuck with previous image. For example can be fixed with this code snippet: `this.srcRules = ESLMediaRuleList.parse(this.src);` in connectedCallback